src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+`;
+
+const Total = styled.div`
+  font-family: 'PTRootUIWebMedium', sans-serif;
+  color: #ff0065;
+  display: contents;
+  font-weight: bold;
+`;
+
+const HorizontalLine = styled.div`
+  margin: 9px 0;
+  width: 100%;
+  height: 2px;
+  background: linear-gradient(90deg, #23a9c9, #cd34a2);
+`;
+
+const Bold = styled.p`
+  font-weight: bold;
+`;
+
+//TODO : change color of block if polygon has less than 10,000 plr on keybased / smart wallet
+const Block = styled.div`
+  font-size: 12px;
+  padding: 2px;
+  text-indent: 2px;
+  display: flex;
+  color: ${(props) => props.color === 'red' && '#ff0065'};
+`;
+
+const mapOfferToOption = (offer: ExchangeOffer) => {
+  const serviceDetails = swapServiceIdToDetails[offer.provider];
+  return {
+    title: serviceDetails?.title ?? offer.provider,
+    value: offer.provider,
+    iconUrl: serviceDetails?.iconUrl,
+  };
+};
+
+// TODO:UNCOMMENT
+const MAX_PLR_TOKEN_LIMIT = 5;
+
+const PlrDaoStakingTransactionBlock = ({
+  id: transactionBlockId,
+  errorMessages,
+  values,
+  multiCallData,
+}: IPlrDaoStakingMembershipBlock) => {
+  const {
+    smartWalletOnly,
+    providerAddress,
+    accountAddress,
+    sdk,
+    getSupportedAssetsWithBalancesForChainId,
+  } = useEtherspot();
+
+  const [amount, setAmount] = useState<string>('');
+  const [selectedOffer, setSelectedOffer] = useState<SelectOption | null>(
+    values?.offer ? mapOfferToOption(values?.offer) : null
+  );
+  const [availableOffers, setAvailableOffers] = useState<
+    ExchangeOffer[] | null
+  >(values?.offer ? [values.offer] : null);
+  const [isLoadingAvailableOffers, setIsLoadingAvailableOffers] =
+    useState<boolean>(false);
+
+  const [selectedFromAsset, setSelectedFromAsset] =
+    useState<IAssetWithBalance | null>(values?.selectedAsset ?? null);
+
+  const [selectedAccountType, setSelectedAccountType] = useState<string>(
+    AccountTypes.Contract
+  );
+
+  const [selectedFromNetwork, setSelectedFromNetwork] = useState<Chain | null>(
+    null
+  );
+  const [keyBasedTotal, setKeyBasedTotal] = useState<number>(0);
+  const [smartWalletTotal, setSmartWalletTotal] = useState<number>(0);
+
+  const [totalKeyBasedPLRTokens, setTotalKeyBasedPLRTokens] = useState<number>(0);
+  const [totalSmartWalletPLRTokens, setTotalSmartWalletPLRTokens] = useState<number>(0);
+
+  const [accounts, setAccounts] = useState<{}[]>([]);
+  const [isNFTMember, setIsNFTMember] = useState<boolean>(false);
+
+  const hasEnoughPLR = totalKeyBasedPLRTokens >= MAX_PLR_TOKEN_LIMIT || totalSmartWalletPLRTokens >= MAX_PLR_TOKEN_LIMIT;
+
+  const fixed = multiCallData?.fixed ?? false;
+  const defaultCustomReceiverAddress =
+    values?.receiverAddress &&
+    !addressesEqual(providerAddress, values?.receiverAddress) &&
+    !addressesEqual(accountAddress, values?.receiverAddress)
+      ? values.receiverAddress
+      : null;
+  const [customReceiverAddress, setCustomReceiverAddress] = useState<
+    string | null
+  >(defaultCustomReceiverAddress);
```

Comment:
please do not break lines when not needed, thanks
```suggestion
  const [customReceiverAddress, setCustomReceiverAddress] = useState<string | null>(defaultCustomReceiverAddress);
```